### PR TITLE
Electrostatic amir

### PR DESCRIPTION
**Pull request by amirbous**
#example

Further work on code provided by @pratikvn going along with progress for using ginkgo in electrostatic

Structure and changes - all changes are done under example/electrostatic-abb:
Added new utils.hpp file to include user defined methods: read input from matrix file, read config file .... .
And moved some of the already existing functions to it
added readme.md for electrstatic specific usage

Purpose
Added Support for binary matrix files: added reader
Added config file (electrto.config) parser , to read parameters for Ginkgo configuration
Ginkgo configuration parameters are set in electro.config. More in the readme page for the electrostatic example

!! Warning !!:
Still uncertain about double and single precision as output files seem to be double precision in ascii format and single precision in binary format. For now when reading binary, reading as float and casting to double.

### CODE REVIEW

**Comment by pratikvn** [comment]
Thanks @amirbous! I think the explicit cast to `double` should not be necessary, but shouldn't hurt. The `matrix_data` object is templated and should implicitly cast to its value_type. 

So, you should be able to automatically run different solvers for both `float` and `double` by changing the `using ValueType = ...` in the `electrostatic_abb.cpp` file and compiling it again.

**Comment by pratikvn** [comment]
@upsj , I think he is not using `gko::write`, but using the write functionality in the casopt library.

**Comment by amirbous** [comment]
@upsj, @pratikvn  Thank you for the check.
The input files I am using are generated externally from another fortran script, seperate from ginkgo, and are using different precision based on the format. I made sure to note it down, so whenever other files are being used.
I will proceed as described by @pratikvn where the user is aware of the datatype being used, and can speicify it in the ValueType.

**Comment by amirbous** [comment]
. To reissue

File: examples/electrostatic-abb/electrostatic-abb.cpp [hunk 1]
```diff
@@ -137,8 +152,6 @@ int main(int argc, char* argv[])
         solver->apply(b, x_clone);
```
Comment (pratikvn) [inline examples/electrostatic-abb/electrostatic-abb.cpp:152]: You can reduce the warmup here in line 150 to 1 instead of 3 to speed things up.

**Review by pratikvn — APPROVED** [review]
This looks okay to me. If I can merge it, let me know.

**Pull request merged**